src/components/input-time-picker/input-time-picker.tsx

```diff
@@ -504,34 +530,38 @@ export class InputTimePicker
           role="combobox"
         >
           <calcite-input
-            disabled={this.disabled}
+            disabled={disabled}
             icon="clock"
             label={getLabelText(this)}
             onCalciteInputInput={this.calciteInputInputHandler}
             onCalciteInternalInputBlur={this.calciteInternalInputBlurHandler}
             onCalciteInternalInputFocus={this.calciteInternalInputFocusHandler}
-            readOnly={this.readOnly}
+            readOnly={readOnly}
             scale={this.scale}
             step={this.step}
             // eslint-disable-next-line react/jsx-sort-props
             ref={this.setCalciteInputEl}
           />
         </div>
         <calcite-popover
+          aria-hidden={toAriaBoolean(!this.open)}
```

Comment:
Since `popover` handles a lot of these roles for us, I think you only need to set `label` here. The rest are [already applied](https://github.com/Esri/calcite-components/blob/master/src/components/popover/popover.tsx#L579-L584).